packages/babel-generator/src/generators/flow.js

```diff
@@ -134,6 +134,96 @@ export function DeclareExportAllDeclaration(/*node: Object*/) {
   ExportAllDeclaration.apply(this, arguments);
 }
 
+export function EnumDeclaration(node: Object) {
+  const { id, body } = node;
+  this.word("enum");
+  this.space();
+  this.print(id, node);
+  this.print(body, node);
+}
+
+function enumExplicitType(
+  context: Object,
+  name: string,
+  hasExplicitType: boolean,
+) {
+  if (hasExplicitType) {
+    context.space();
+    context.word("of");
+    context.space();
+    context.word(name);
+  }
+  context.space();
+}
+
+function enumBody(context: Object, node: Object) {
+  const { members } = node;
+  context.token("{");
+  context.indent();
+  context.newline();
+  for (const member of members) {
+    context.print(member, node);
+    context.newline();
+  }
+  context.dedent();
+  context.token("}");
+}
+
+export function EnumBooleanBody(node: Object) {
+  const { explicitType } = node;
+  enumExplicitType(this, "boolean", explicitType);
+  enumBody(this, node);
+}
+
+export function EnumNumberBody(node: Object) {
+  const { explicitType } = node;
+  enumExplicitType(this, "number", explicitType);
+  enumBody(this, node);
+}
+
+export function EnumStringBody(node: Object) {
+  const { explicitType } = node;
+  enumExplicitType(this, "string", explicitType);
+  enumBody(this, node);
+}
+
+export function EnumSymbolBody(node: Object) {
+  enumExplicitType(this, "symbol", true);
+  enumBody(this, node);
+}
+
+export function EnumDefaultedMember(node: Object) {
+  const { id } = node;
+  this.print(id, node);
+  this.token(",");
+}
+
+function enumInitializedMember(context: Object, node: Object) {
+  const { id, init } = node;
+  context.print(id, node);
+  context.space();
+  context.token("=");
+  context.space();
+  if (typeof init === "boolean") {
```

Comment:
When will `init` be a boolean? I guess they are always an AST node?

Reply:
Ah yes, thanks for pointing this out. It used to be able to be a `boolean` but I updated this to use a `BooleanLiteral` node after a comment from Nicolò